runtime/tests/org.eclipse.core.tests.runtime/src/org/eclipse/core/tests/runtime/jobs/JobTest.java

```diff
@@ -1602,6 +1620,38 @@ protected IStatus run(IProgressMonitor monitor) {
 		group.done();
 	}
 
+	public void testSetProgressGroup_progressGroupIsCanceled_monitorInJobIsCanceled() throws InterruptedException {
+		Job checkMonitorIsCanceled = createJobThatChecksItsOwnMonitor();
+
+		final IProgressMonitor progressGroup = Job.getJobManager().createProgressGroup();
+		progressGroup.beginTask("", 2);
+
+		checkMonitorIsCanceled.setProgressGroup(progressGroup, 1);
+
+		// let the job do the cancellation
```

Comment:
I do not understand that comment as the job does not cancel anything. I don't think that a comment is necessary for comprehension.